react-components/src/data-providers/core-dm-provider/getCadConnectionsForRevisions.ts

```diff
@@ -0,0 +1,194 @@
+      cadNodeProps.model3D.space === modelRef.space
+  );
+
+  if (modelRevisionPair === undefined) {
+    return undefined;
+  }
+
+  const revisionIndex = cadNodeProps.revisions.findIndex(
+    (propRevision) =>
+      modelRevisionPair[1].externalId === propRevision.externalId &&
+      modelRevisionPair[1].space === propRevision.space
+  );
+
+  if (revisionIndex === -1) {
+    return undefined;
+  }
+
+  const modelId = getModelIdFromExternalId(cadNodeProps.model3D.externalId);
+  const revisionId = getRevisionIdFromExternalId(cadNodeProps.revisions[revisionIndex].externalId);
+  const treeIndex = cadNodeProps.treeIndexes[revisionIndex];
+
+  return { modelId, revisionId, treeIndex };
+}
+
+type SourcesSelectType = [
+  { source: typeof COGNITE_CAD_NODE_SOURCE; properties: CogniteCADNodeProperties },
+  { source: typeof COGNITE_ASSET_SOURCE; properties: CogniteAssetProperties }
+];
+
+async function getModelConnectionResults(
+  modelRevisions: Array<[DmsUniqueIdentifier, DmsUniqueIdentifier]>,
+  fdmSdk: FdmSDK
+): Promise<QueryResult<typeof cadConnectionsQuery, SourcesSelectType>> {
+  const parameters = {
+    modelRefs: modelRevisions.map(([modelRef, _revisionRef]) => restrictToDmsId(modelRef)),
+    revisionRefs: modelRevisions.map(([_modelRef, revisionRef]) => restrictToDmsId(revisionRef))
+  };
+
+  const query = {
+    ...cadConnectionsQuery,
+    parameters
+  };
+
+  return await fdmSdk.queryAllNodesAndEdges<typeof query, SourcesSelectType>(query);
+}
+
+const cadConnectionsQuery = {
+  with: {
+    cad_nodes: {
+      nodes: {
+        filter: {
+          and: [
+            {
+              in: {
+                property: [
+                  CORE_DM_3D_CONTAINER_SPACE,
+                  COGNITE_CAD_NODE_SOURCE.externalId,
+                  'model3D'
+                ],
+                values: { parameter: 'modelRefs' }
+              }
+            },
+            {
+              containsAny: {
+                property: [
+                  CORE_DM_3D_CONTAINER_SPACE,
+                  COGNITE_CAD_NODE_SOURCE.externalId,
+                  'revisions'
+                ],
+                values: { parameter: 'revisionRefs' }
+              }
+            }
+          ]
+        }
+      },
+      limit: 10000
+    },
+    object_3ds: {
+      nodes: {
+        from: 'cad_nodes',
+        through: { view: COGNITE_CAD_NODE_SOURCE, identifier: 'object3D' },
+        direction: 'outwards',
+        filter: {
+          hasData: [COGNITE_3D_OBJECT_SOURCE]
+        }
+      },
+      limit: 10000
+    },
+    assets: {
+      nodes: {
+        from: 'object_3ds',
+        through: { view: COGNITE_ASSET_SOURCE, identifier: 'object3D' },
+        direction: 'inwards'
+      },
+      limit: 10000
+    }
+  },
+  select: {
+    cad_nodes: { sources: cogniteCadNodeSourceWithProperties },
+    assets: { sources: cogniteAssetSourceWithProperties }
+  }
+} as const satisfies Omit<QueryRequest, 'cursor' | 'parameters'>;
```

Comment:
Is it possible (and if does make sense) to move out the consts and the type definitions? Or at least organize into a separated place inside the file. I think moving out we can reduce the file size.

Reply:
Agreed, moved it out :+1: